pkg/volume/util/resize_util.go

```diff
@@ -220,8 +224,7 @@ func MarkFSResizeFinished(
 
 	// if RecoverVolumeExpansionFailure is enabled, we need to reset ResizeStatus back to nil
 	if utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
-		expansionFinished := v1.PersistentVolumeClaimNoExpansionInProgress
-		newPVC.Status.ResizeStatus = &expansionFinished
+		newPVC.Status.AllocatedResourceStatus = nil
```

Comment:
This should only delete the `storage` entry.

Reply:
fixed.